### DISCUSSION TITLE
Looking to Contribute: Some Questions I Have

### DISCUSSION THREAD

**Discussion by ghost** [saivert/pwvucontrol#59] [Q&A]
Hello, 

I'm looking to contribute to this project as I use this application quite often on my system, and had a few questions.

Any issues you'd like prioritized or think would be a good first issue to take a look at?

Also why did you choose to use Meson instead of just using the built in Cargo tool for Rust? I see you had C/C++ background so I'm wondering if it was just what you're used to. Are you open to the idea of switching to just fully use Cargo instead of Meson I was going to try and do that swap first if that's fine with you. (unless I'm wrong on my idea of what Cargo can handle properly I'm still learning the code base and the build)

I see you wanted more code robustness, would you rather I spend some time trying to add better error handling?

Thanks,
kittytree.

**Comment by saivert** [comment]
Meson is used to build the non rust parts like the locale stuff and compile the resources used by GTK. Meson has excellent tooling for that and it is how most of the GTK projects do it. Then it invokes cargo to build the rust code.

Feel free to contribute to whatever you feel like.  Better error handling is a good first step.

**Reply by ninetailedtori** [reply]
Is it still worth contributing right now? The project has quite a few issues that seem to be unhandled, are you looking for more maintainers to oversee PRs and handle bug reports etc? /genuine

**Reply by saivert** [reply]
Not sure what contributions you had in mind but I would be happy to receive more code contributions at least. I don't need help managing issues or any administrative tasks. The major pain point is that this project depends on `wireplumber-rs` project which isn't kept up to date as well as I would like. If someone really wants to rewrite this to get rid of the `wireplumber-rs` dependency and just use direct pipewire API it is fine by me. It is totally doable but you will basically have to recreate some of the management logic that wireplumber offers and which is why I went with it after initially trying to use pipewire API only.

I'm a full time software developer right now so I have very little time (and energy) for open source work.
